compiler/llvm/clangUtil.cpp

```diff
@@ -4444,13 +4444,38 @@ static void moveResultFromTmp(const char* resultName, const char* tmpbinname);
 static llvm::CodeGenFileType getCodeGenFileType() {
   switch (getGpuCodegenType()) {
     case GpuCodegenType::GPU_CG_AMD_HIP:
-      return llvm::CodeGenFileType::CGFT_ObjectFile;
     case GpuCodegenType::GPU_CG_NVIDIA_CUDA:
     default:
       return llvm::CodeGenFileType::CGFT_AssemblyFile;
   }
 }
 
+static std::string findSiblingClangToolPath(const std::string &toolName) {
+  // Find path to a tool that is a sibling to clang
+  // note that if we have /path/to/clang-14, this logic
+  // will loop for /path/to/${toolName}-14.
+  //
+  // If such suffixes do not turn out to matter in practice, it would
+  // be nice to update this code to use sys::path::parent_path().
+  std::vector<std::string> split;
+  std::string result = toolName;
+
+  splitStringWhitespace(CHPL_LLVM_CLANG_C, split);
+  if (split.size() > 0) {
+    std::string tmp = split[0];
+    const char* clang = "clang";
+    auto pos = tmp.find(clang);
+    if (pos != std::string::npos) {
+      tmp.replace(pos, strlen(clang), toolName);
+      if (pathExists(tmp.c_str())) {
+        result = tmp;
+      }
+    }
+  }
+
```

Comment:
Should we set result to empty string and `INT_FATAL` here if it is still empty? Relatedly, this returns the toolName directly if it can't find anything, should we also check whether the tool is in PATH and readily available (but that's probably shaky because of version mismatch you talk about in the OP?)?

Reply:
Uh-oh, now seeing that this bit existed before. I understand if you're less enthusiastic of changing it because of that.